articles/per_package_documentation.md

```diff
@@ -0,0 +1,99 @@
+---
+layout: default
+title: Per-Package Documentation
+permalink: articles/per_package_documentation.html
+abstract: This article describes the requirements and design of ROS 2’s per-package documentation system.
+author: Marya Belanger
+published: true
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Background
+
+ROS 2 is lacking a process for the conglomeration of package documentation.
+Discoverability for package documentation (and documentation in general) is one of the most frequent user complaints.
+
+With a standardized package documentation process that encourages package maintainers to robustly document their packages, and presents the results alongside ROS 2's general documentation, issues of discoverability and consistency will be addressed.
+
+In general, our vision for the system is:
+
+- Package maintainers document their packages (API docs, manually written content, etc) in their packages' repositories following some guidelines or templates we recommend
+- Package documentation from the repositories is built on our documentation site by our buildfarm in an automatic process with no extra input from the maintainers
```

Comment:
Does no extra input mean nothing beyond adding a `doc` entry in the ROS distro's `distribution.yaml` to opt in?

Reply:
Yeah, I think that this could be clarified a bit with:

```
Package documentation from the repositories is built on the ROS 2 infrastructure and deployed to the ROS 2 documentation site in an automatic process with no input from the maintainer besides an opt-in `doc` block in the distribution.yaml
```

Also, we should probably put a link to [REP 141](https://www.ros.org/reps/rep-0141.html) for context.